### ISSUE TITLE
Quick filters with object as default value remain visible when removing filter

### ISSUE THREAD

**Issue by nicgirault** [marmelab/react-admin#8478]
**What you were expecting:**

When dealing with dates, it can be useful to create custom (quick) filters. Our backend data provider support all the ORM (mongoose) operators so I want to use filters like this:

```
const QuickFilter = ({ label, source, defaultValue }) => {
  const translate = useTranslate();
  return <Chip sx={{ marginBottom: 1 }} label={translate(label)} />;
};

const postFilter = [
  <QuickFilter
    label="Published in 2022"
    source="publishedAt"
    defaultValue={{ $gt: new Date(2022), $lt: new Date(2023) }}
  />
];
```

The filtering behavior is working correctly but when clicking on the hide button on the left of the chip, the filter remains visible.

The default value is an object and is the reason of the bug (no bug with a string for example).

**What happened instead:**

The filter is supposed to be removed.

**Steps to reproduce:**

https://codesandbox.io/s/compassionate-forest-myfqni?file=/src/posts/PostList.tsx

**Environment**

* React-admin version: 4.6

**Comment by septentrion-730n** [comment]
Thanks for your report, confirmed!

**Comment by slax57** [comment]
Fixed by #8637
